waiter/src/waiter/auth/authentication.clj

```diff
@@ -162,20 +188,21 @@
   Authenticator
   (wrap-auth-handler [_ request-handler]
     (let [single-user-prefix "SingleUser "]
-      (fn anonymous-handler [request]
+      (fn anonymous-handler [{:keys [headers] :as request}]
         (let [auth-header (select-auth-header request #(str/starts-with? % single-user-prefix))
+              {:strs [x-waiter-single-user]} headers
```

Comment:
Please document the purpose of `x-waiter-single-user` and the general reason for changes to this function.

Reply:
The comment was helpful.  Thank you.